common/src/main/java/gov/cms/ab2d/common/health/MemoryUtilization.java

```diff
@@ -0,0 +1,29 @@
+package gov.cms.ab2d.common.health;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+/**
+ * Health check on memory - make sure you aren't running to low
+ */
+public class MemoryUtilization {
+    /**
+     * Make sure you can allocate memory of at least a certain number of MB. If you get an out
+     * of memory error, you can't so return true - you're out of memory.
+     *
+     * This is done with a two dimensional array. It's an array of one MB arrays of bytes. I split it up
+     * so that we didn't have to have one contiguous set of memory
+     *
+     * @param numMBToCreate - the number of megabytes to create
+     * @return if the JVM runs out of memory while trying to allocate the array
+     */
+    public static boolean outOfMemory(int numMBToCreate) {
+        try {
+            byte[][] arr = new byte[numMBToCreate][1024];
```

Comment:
Let's actually fill the array with random bytes. Who knows what kind of optimizations JVM undertakes when initializing (but not yet writing to) an array; may be it does _lazy init_ of some kind 🤷‍♂ . Switching to one-dimensional array will make this easier, too.

Reply:
I actually ran into that problem with a one-dimensional array.

Reply:
Ah. Max array size is the max Integer. Makes sense.

Reply:
The one-dimensional array actually seemed to be optimized and I couldn't get it to fail. That's why I chose the 2 dimensional.

Reply:
Yeah. A one-dimensional byte array can't be larger than ~2GB, you likely have more memory available.